README.md
# Humphrey Child Theme

## Installation
1. Upload the Humphrey child theme folder via FTP to your wp-content/themes/ directory. (The Genesis parent theme needs to be in the wp-content/themes/ directory as well.)

2. Go to your WordPress dashboard and select Appearance.

3. Activate the Humphrey theme.

4. Inside your WordPress dashboard, go to Genesis > Theme Settings and configure them to your liking.

## Usage

### Widgets

The homepage features 4 custom widget areas:

1. Home Slider. For use with the Genesis Responsive Slider plugin.

2. Home Welcome. Use a text widget here to display a welcome message etc.

3. Home Left. Use the Genesis Featured Posts widget here.

4. Home Right. You can use a variety of widgets here such as a text widget, Genesis Latest Tweets widget, Genesis eNews Extended widget etc.

Humphrey contains 2 additional custom widget areas.

1. After Post Block. This widget area appears after a post on all single post pages. You can use a text widget or even the Genesis eNews Extended widget here.

2. 404 Error Page. The 404 error page is completely widgetized. You can add all sorts of widgets here as seen on the demo site.

## Configuring the widgets as seen on the demo site

### Genesis Responsive Slider

1. Maximum Slider Width = 1100.
2. Maximum Slider Height = 350.
3. Check Display Next / Previous Arrows in Slider?
4. Check Display Pagination in Slider?
5. Check Display Post/Page Title in Slider?
6. Check Display Content in Slider?
7. Select Display post content from drop menu.
8. Enter 150 into Limit content to text field.
9. Slider Excerpt Width = 50
10. Excerpt Location vertical = Bottom
11. Excerpt Location horizontal = Left

### Genesis Featured Posts

1. Check Show Featured Image
2. Select Image Size homepage 150x150 from drop menu
3. Select Left for Image Alignment
4. Check Show Post Title. Do not check Show Post Info
5. Select Show Content Limit for Content Type from drop menu
6. Enter 300 into Limit content to text field

The rest of the widgets are self-evident I think.

## Portfolio Page
The Portfolio Page makes use of a Custom Post Type called Portfolio. To add items to your Portfolio Page:

1. From the menu click on Portfolio > Add New
2. Add a short title
3. Upload an image to the post. You must also set this image as Feature Image
4. Add a short description

On the demo site I uploaded large images with the Max Width being set as 1024 as per the settings at Settings > Media. These are displayed on the Portfolio single post pages.
The theme creates the 340 x 230 image size which is displayed on the Portfolio category page. Be sure to re-save your permalinks or the Custom Post Type may not display.

## Image Sizes
In addition to the image sizes created by WordPress as per your settings at Settings > Media, Humphrey comes with the following image sizes as defined in the `functions.php` file:

1. homepage (120 x 120)
2. small (80 x 80)
3. portfolio (340 x 230)

You will probably have to regenerate your images for them to display properly. Refer to the section below called Recommended Plugins for details.

## Custom Content Boxes
The custom content boxes make use of DIV classes. For example,

`<div class="content-box-blue">some random text</div>`

There are 6 styles to choose from. Refer to the style sheet for details.

## Gradient Buttons
Each button has a unique class.

`<a class="button-green" href="#">Link Text</a>`

The above will render the Green button. There are 11 styles to choose from. Refer to the style sheet for details.

## Post Formats
Humphrey comes with support for the following Post Formats:

1. Aside
2. Audio
3. Chat
4. Gallery
5. Image
6. Link
7. Quote
8. Status
9. Video

To use them simply select the appropriate Format from the Post Format metabox in the Add New Post screen.

Further reading on Post Formats at http://codex.wordpress.org/Post_Formats

## Recommended Plugins
1. [Genesis Responsive Slider](http://wordpress.org/extend/plugins/genesis-responsive-slider/)
2. [Genesis eNews Extended](http://wordpress.org/extend/plugins/genesis-enews-extended/)
3. [Regenerate Thumbnails](http://wordpress.org/extend/plugins/regenerate-thumbnails/)

## Credit
1. Content Boxes by Brian Gardner - http://www.briangardner.com/genesis-content-boxes/
2. Custom Post Type Portfolio Page by Brian Gardner - http://www.briangardner.com/portfolio-page/
3. Gradient Buttons by Brian Gardner - http://www.briangardner.com/social-media-buttons/
4. Widgetized 404 Page by David Decker - http://genesisthemes.de/en/2011-08/tutorial-widgetized-404-error-page-in-genesis/
5. Icons for eNews input fields by iconmonstr - http://iconmonstr.com/
6. Post Format icons by StudioPress - http://www.studiopress.com/graphics/post-format-icons

## Further
1. [Theme Release Page](http://wpcanada.ca/our-themes/humphrey/)
2. [Theme Demo](http://demo.wpcanada.ca/humphrey/)
3. Download from [Google](http://code.google.com/p/humphrey-theme/)
4. Download from [GitHub](https://github.com/lenkutchma/humphrey)

## Bug Reports
1. [Google](http://code.google.com/p/humphrey-theme/issues/list)
2. [GitHub](https://github.com/lenkutchma/humphrey/issues)

Enjoy!

changelog.txt
*** Humphrey Changelog ***


2011.01.27 - Version 1.0

  * changelog.txt - Added changelog and revised version number

2012.02.04 - version 1.1
  * added missing code to functions.php to activate Genesis Box
  * added styling to style.css for Genesis Box
  * added Genesis image to images folder for Genesis Box
  * updated changelog and revised version number

2012.02.06 - version 1.2
  * added Post Formats
  * updated changelog and version number

2012.02.09 - version 1.3.0
  * added internationalization support
  * updated changelog and version number

2012.07.07 - version 1.4.0
  * cleaned up style.css in accordance with Genesis CSS Best Practices == http://www.briangardner.com/code/css-best-practices/
  * restyled Alert Boxes and renamed them Content Boxes. I prefer Brian's naming convention == http://www.briangardner.com/genesis-content-boxes/
  * updated changelog
  * updated README

05 FEB 2013 - Version 2.0
  * complete theme re-write

07 FEB 2013 - Version 2.0.1
  * minor updates to header widget area menu and latest tweets widget

03 AUG 2013 - version 2.0.2
  * fixed translation problem

404.php
<?php
/**
 * 404 Not found page example
 * 
 * This helps create a custom and widgetized 404 error page template for Genesis child themes.
 *
 * @author David Decker, http://genesisthemes.de/en/
 * @link http://genesisthemes.de/en/2011-08/tutorial-widgetized-404-error-page-in-genesis/
 *
 * Note by Len, added class 'entry-content' so that styling for 404 page is inherited by theme.
 */


remove_action( 'genesis_loop', 'genesis_do_loop' );
add_action( 'genesis_loop', 'humphrey_404_loop_helper' );
/**
 * Add widget support for the 404 page. If no widgets active, display the default loop/message.
 *
 */
function humphrey_404_loop_helper() {

	if ( is_active_sidebar( '404-page' ) ) {
			echo '<div class="entry-content">';
		dynamic_sidebar( '404-page' );
			echo '</div><!-- end .entry-content -->';		
	}
	else {
		genesis_standard_loop();
	}
	
}

genesis();

home.php
<?php

add_action( 'genesis_meta', 'humphrey_home_genesis_meta' );
/**
 * Add widget support for homepage. If no widgets active, display the default loop.
 *
 */
function humphrey_home_genesis_meta() {

	if ( is_active_sidebar( 'home-slider' ) || is_active_sidebar( 'home-welcome' ) || is_active_sidebar( 'home-left' ) || is_active_sidebar( 'home-right' ) ) {

		remove_action( 'genesis_loop', 'genesis_do_loop' );
		add_action( 'genesis_loop', 'humphrey_home_loop_helper' );
		add_filter( 'genesis_pre_get_option_site_layout', '__genesis_return_full_width_content' );

		// Add humphrey-home body class
		add_filter( 'body_class', 'humphrey_body_class' );
		function humphrey_body_class( $classes ) {
   			$classes[] = 'humphrey-home';
  			return $classes;
		}

	}
}

/**
 * Display widget content for "home" widget sections
 *
 */

function humphrey_home_loop_helper() {

		if ( is_active_sidebar( 'home-slider' ) ) {
			echo '<div id="homepage-slider">';
			dynamic_sidebar( 'home-slider' );
			echo '</div><!-- end #homepage-slider -->';
		}

		if ( is_active_sidebar( 'home-welcome' ) ) {
			echo '<div id="homepage-welcome">';
			dynamic_sidebar( 'home-welcome' );
			echo '</div><!-- end #homepage-welcome -->';
		}

		if ( is_active_sidebar( 'home-left' ) ) {
			echo '<div id="homepage-left">';
			dynamic_sidebar( 'home-left' );
			echo '</div><!-- end #homepage-left -->';
		}

		if ( is_active_sidebar( 'home-right' ) ) {
			echo '<div id="homepage-right" class="widget-area">';
			dynamic_sidebar( 'home-right' );
			echo '</div><!-- end #homepage-right -->';
		}
}

genesis();

archive-portfolio.php
<?php
/**
 * The custom portfolio post type archive template
 */

/** Force full width content layout */
add_filter( 'genesis_pre_get_option_site_layout', '__genesis_return_full_width_content' );

/** Remove the post info function */
remove_action( 'genesis_before_post_content', 'genesis_post_info' );

/** Remove the post content */
remove_action( 'genesis_post_content', 'genesis_do_post_content' );

/** Remove the post image */
remove_action( 'genesis_post_content', 'genesis_do_post_image' );

/** Add the featured image after post title */
add_action( 'genesis_post_title', 'humphrey_portfolio_grid' );
function humphrey_portfolio_grid() {
	if ( has_post_thumbnail() ){
		echo '<div class="portfolio-featured-image">';
		echo '<a href="' . get_permalink() .'" title="' . the_title_attribute( 'echo=0' ) . '">';
		echo get_the_post_thumbnail($thumbnail->ID, 'portfolio' );
		echo '</a>';
		echo '</div>';
	}
}

/** Remove the post meta function */
remove_action( 'genesis_after_post_content', 'genesis_post_meta' );

genesis();

functions.php
<?php
// Start the engine
require_once( get_template_directory() . '/lib/init.php' );

// Localization
load_child_theme_textdomain( 'humphrey', get_stylesheet_directory() . '/lib/languages' );

// Child theme (do not remove)
define( 'CHILD_THEME_NAME', 'Humphrey Theme' );
define( 'CHILD_THEME_URL', 'http://wpcanada.ca/our-themes/humphrey' );

// Add Viewport meta tag for mobile browsers
add_action( 'genesis_meta', 'humphrey_viewport_meta_tag' );
function humphrey_viewport_meta_tag() {
	echo '<meta name="viewport" content="width=device-width, initial-scale=1.0"/>';
}

// Load Google fonts
add_action( 'wp_enqueue_scripts', 'humphrey_load_google_font' );
function humphrey_load_google_font() {
	wp_enqueue_style( 'google-font', 'http://fonts.googleapis.com/css?family=Oswald:400|Open+Sans:400', array(), PARENT_THEME_VERSION );
}

// Add support for custom background
add_theme_support( 'custom-background' );

// Add support for custom header
add_theme_support( 'genesis-custom-header', array(
	'width' => 1100,
	'height' => 120
) );

// Reposition the secondary navigation
remove_action( 'genesis_after_header', 'genesis_do_subnav' );
add_action( 'genesis_before', 'genesis_do_subnav' );

// Add support for 3-column footer widgets
add_theme_support( 'genesis-footer-widgets', 3 );

// Add new image sizes
add_image_size( 'homepage', 120, 120, TRUE );
add_image_size( 'small', 80, 80, TRUE );
add_image_size( 'portfolio', 340, 230, TRUE );

// Add single post navigation
add_action( 'genesis_before_comments', 'humphrey_post_nav' );
function humphrey_post_nav(){
	if (is_single()) {
	echo '<div class="post-nav">';
	echo '<div class="next-post-nav">';
	echo '<span class="next">';
	echo (__( 'Next Article', 'humphrey' ));
	echo '</span>';
	echo next_post_link('%link', '%title');
	echo '</div>';
	echo '<div class="prev-post-nav">';
	echo '<span class="prev">';
	echo (__( 'Previous Article', 'humphrey' ));
	echo '</span>';
	echo previous_post_link('%link', '%title');
	echo '</div>';
	echo '</div>';
	}
}

// Add after post block section to single post page
add_action( 'genesis_after_post_content', 'humphrey_after_post_block'  ); 
function humphrey_after_post_block() {
	if ( is_single() && is_active_sidebar( 'after-post-block' ) ) {
	echo '<div class="after-post-block"><div class="wrap">';
	dynamic_sidebar( 'after-post-block' );
	echo '</div></div>';
	}
}

// Create portfolio custom post type
add_action( 'init', 'humphrey_portfolio_post_type' );
function humphrey_portfolio_post_type() {
	register_post_type( 'portfolio',
		array(
			'labels' => array(
				'name' => __( 'Portfolio', 'humphrey' ),
				'singular_name' => __( 'Portfolio', 'humphrey' ),
			),
			'exclude_from_search' => true,
			'has_archive' => true,
			'hierarchical' => true,
			'menu_icon' => get_stylesheet_directory_uri() . '/images/icons/portfolio.png',
			'public' => true,
			'rewrite' => array( 'slug' => 'portfolio' ),
			'supports' => array( 'title', 'editor', 'author', 'thumbnail', 'excerpt', 'trackbacks', 'custom-fields', 'comments', 'revisions', 'page-attributes', 'genesis-seo' ),
		)
	);
}

// Change the number of portfolio items to be displayed (props Bill Erickson)
add_action( 'pre_get_posts', 'humphrey_portfolio_items' );
function humphrey_portfolio_items( $query ) {

	if( $query->is_main_query() && !is_admin() && is_post_type_archive( 'portfolio' ) ) {
		$query->set( 'posts_per_page', '12' );
	}

}

// Add support for post formats
add_theme_support( 'post-formats', array(
	'aside',
	'audio',
	'chat',
	'gallery',
	'image',
	'link',
	'quote',
	'status',
	'video'
) );

add_theme_support( 'genesis-post-format-images' );

// Remove elements for post formats
add_action( 'genesis_before_post', 'humphrey_remove_elements' );
function humphrey_remove_elements() {
	
	if ( ! current_theme_supports( 'post-formats' ) )
		return;

	// Remove if post has format
	if ( get_post_format() ) {
		remove_action( 'genesis_post_title', 'genesis_do_post_title' );
		remove_action( 'genesis_before_post_content', 'genesis_post_info' );
		remove_action( 'genesis_after_post_content', 'genesis_post_meta' );
	}
}

// Reposition the footer widgets
remove_action( 'genesis_before_footer', 'genesis_footer_widget_areas' );
add_action( 'genesis_after', 'genesis_footer_widget_areas' );

// Reposition the footer
remove_action( 'genesis_footer', 'genesis_footer_markup_open', 5 );
remove_action( 'genesis_footer', 'genesis_do_footer' );
remove_action( 'genesis_footer', 'genesis_footer_markup_close', 15 );
add_action( 'genesis_after', 'genesis_footer_markup_open', 11 );
add_action( 'genesis_after', 'genesis_do_footer', 12 );
add_action( 'genesis_after', 'genesis_footer_markup_close', 13 );

// Register widget areas
genesis_register_sidebar( array(
	'id'			=> 'home-slider',
	'name'			=> __( 'Home Slider', 'humphrey' ),
	'description'		=> __( 'This is the homepage slider section.', 'humphrey' ),
) );
genesis_register_sidebar( array(
	'id'			=> 'home-welcome',
	'name'			=> __( 'Home Welcome', 'humphrey' ),
	'description'		=> __( 'This is the homepage welcome section.', 'humphrey' ),
) );
genesis_register_sidebar( array(
	'id'			=> 'home-left',
	'name'			=> __( 'Home Left', 'humphrey' ),
	'description'		=> __( 'This is the homepage left section.', 'humphrey' ),
) );
genesis_register_sidebar( array(
	'id'			=> 'home-right',
	'name'			=> __( 'Home Right', 'humphrey' ),
	'description'		=> __( 'This is the homepage right section.', 'humphrey' ),
) );
genesis_register_sidebar( array(
	'id'			=> 'after-post-block',
	'name'			=> __( 'After Post Block', 'humphrey' ),
	'description'		=> __( 'This is the after post block section.', 'humphrey' ),
) );
genesis_register_sidebar( array(
	'id'			=> '404-page',
	'name'			=> __( '404 Page', 'humphrey' ),
	'description'		=> __( 'This is the widget area of the 404 Not Found Page Template.', 'humphrey' ),
) );